core/src/test/java/org/infinispan/manager/AllClusterExecutorTest.java

```diff
@@ -403,6 +404,9 @@ public void call() throws InterruptedException, ExecutionException, TimeoutExcep
             Exceptions.expectExecutionException(org.infinispan.util.concurrent.TimeoutException.class, futureRemote);
             checkPoint.get().awaitStrict("block_execution", 10, TimeUnit.SECONDS);
             checkPoint.get().trigger("resume_execution");
+            // Have to wait for callback to complete - otherwise a different thread could find the "resume_execution"
+            // checkpoint reached incorrectly
+            checkPoint.get().awaitStrict("complete", 10, TimeUnit.SECONDS);
```

Comment:
@wburns I think the comment is unnecessarily vague about which "different thread" is going to find the checkpoint, and also the name "complete" sounds wrong because the callback is still not done at the point the checkpoint is triggered. Maybe it would have been simpler to just use a different callback for the second `submitConsumer` call, with different event names?

Reply:
Depends upon your definition of simpler, this change only requires 3 lines, the other would require quite a few more.

The reason for the ambiguity of "different thread" is because we don't really know what thread the `AllClusterExecutorTest` may use at this point. Also the callback is effectively done when triggering this checkpoint as there is no more code for the lambda to invoke.

Reply:
I meant even if it requires more changes, the final result would be simpler, because different test methods cannot influence each other. Isn't the uncertainty about which thread consumes the event caused solely by callback from earlier potentially still running?